### PR TITLE
simplified some expressions, redundant expressions removed

### PR DESCRIPTION
**Pull request by anton-erofeev**

### CODE REVIEW

**Review by filiphr — CHANGES_REQUESTED** [review]
Thanks for this cleanups @anton-erofeev. I only have one small remark, see my comment

File: processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java [hunk 1]
```diff
@@ -498,12 +496,8 @@ private boolean isCorrectlySealed(Type mappingSourceType) {
...
                     new ArrayList<>( mappingSourceType.getPermittedSubclasses() );
                 method.getOptions().getSubclassMappings().forEach( subClassOption -> {
-                    for (Iterator<? extends TypeMirror> iterator = unusedPermittedSubclasses.iterator();
-                                   iterator.hasNext(); ) {
-                        if ( ctx.getTypeUtils().isSameType( iterator.next(), subClassOption.getSource() ) ) {
-                            iterator.remove();
-                        }
-                    }
+                    unusedPermittedSubclasses.removeIf(
+                            typeMirror -> ctx.getTypeUtils().isSameType(typeMirror, subClassOption.getSource()));
```
[the remark this answers is not in the document]
Comment (anton-erofeev) [reply processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java]: ok, I reverted this change, thanks

**Comment by filiphr** [comment]
@anton-erofeev seems like there are some formatting issues. Can you have a look at it please?

**Comment by anton-erofeev** [comment]
> @anton-erofeev seems like there are some formatting issues. Can you have a look at it please?
updated but it looks like I can't run checks

**Comment by filiphr** [comment]
You are contributing for the first time @anton-erofeev and that's why the checks don't run automatically. I need to approve them in order for them to run. Just did that, if everything is OK, we'll merge this. For your next contributions this won't be necessary.

File: processor/src/test/java/org/mapstruct/ap/test/nestedbeans/HouseDto.java [hunk 2]
```diff
@@ -58,10 +60,10 @@ public boolean equals(Object o) {
         if ( year != houseDto.year ) {
             return false;
         }
-        if ( name != null ? !name.equals( houseDto.name ) : houseDto.name != null ) {
+        if ( !Objects.equals( name, houseDto.name ) ) {
             return false;
         }
-        return roof != null ? roof.equals( houseDto.roof ) : houseDto.roof == null;
+        return Objects.equals(roof, houseDto.roof);
```
Comment (filiphr) [inline processor/src/test/java/org/mapstruct/ap/test/nestedbeans/HouseDto.java:66]:
```suggestion
        return Objects.equals( roof, houseDto.roof );
```

**Comment by filiphr** [comment]
thank @anton-erofeev

**Pull request merged**